lib/saluki-env/src/autodiscovery/providers/local.rs

```diff
@@ -0,0 +1,332 @@
+        cluster_check: false,
+        node_name: MetaString::empty(),
+        source: MetaString::from_static("local"),
+        ignore_autodiscovery_tags: false,
+        metrics_excluded: false,
+        logs_excluded: false,
+        advanced_ad_identifiers: Vec::new(),
+    };
+
+    Ok((config_id, config))
+}
+
+/// Scan and emit events based on configuration files in the directory
+async fn scan_and_emit_events(
+    paths: &[PathBuf], known_configs: &mut HashSet<String>, sender: &Sender<AutodiscoveryEvent>,
+) -> Result<(), GenericError> {
+    let mut found_configs = HashSet::new();
+
+    for path in paths {
+        let mut entries = fs::read_dir(path).await?;
+        while let Ok(Some(entry)) = entries.next_entry().await {
+            let path = entry.path();
+
+            // Only process YAML files
+            if let Some(ext) = path.extension() {
+                if ext == "yaml" || ext == "yml" {
+                    // Process the file if it's a valid configuration
+                    match parse_config_file(&path).await {
+                        Ok((config_id, config)) => {
+                            found_configs.insert(config_id.clone());
+
+                            // Check if this is a new or updated configuration
+                            if !known_configs.contains(&config_id) {
+                                debug!("New configuration found: {}", config_id);
+
+                                let event = AutodiscoveryEvent::Schedule { config };
+                                emit_event(sender, event);
+                                known_configs.insert(config_id.clone());
+                            } else {
+                                // Config ID exists, but the content might have changed
+                                debug!("Configuration updated: {}", config_id);
+
+                                let event = AutodiscoveryEvent::Schedule { config };
+                                emit_event(sender, event);
+                            }
+                        }
+                        Err(e) => {
+                            warn!("Failed to parse config file {}: {}", path.display(), e);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Clean up removed configurations
+    let to_remove: Vec<String> = known_configs
+        .iter()
+        .filter(|config_id| !found_configs.contains(*config_id))
+        .cloned()
+        .collect();
+
+    for config_id in to_remove {
+        debug!("Configuration removed: {}", config_id);
+        known_configs.remove(&config_id);
+
+        // Create an unschedule Config event
+        let event = AutodiscoveryEvent::Unscheduled { config_id };
+        emit_event(sender, event);
+    }
+
+    Ok(())
+}
+
+fn emit_event(sender: &Sender<AutodiscoveryEvent>, event: AutodiscoveryEvent) {
+    match sender.send(event) {
+        Ok(_) => (),
+        Err(e) => {
```

Comment:
An error during send on a broadcast channel implies there are no active receivers, so you can effectively ignore them.